lib/game.rb

```diff
@@ -31,123 +35,102 @@ def self.count_of_games_by_season
 
 #deliverable
   def self.average_goals_per_game
-    sum = @@all.sum { |game| game.away_goals + game.home_goals}.to_f
-    (sum / @@all.length.to_f).round(2)
-  end
-
-  def self.games_per(csv_header)
-    #returns number of games in a given (:season) or (:away_team_id)...
-    group_by_header = @@all.group_by { |game| game.send(csv_header) }
-    group_by_header.values.map{ |games| games.length}
-  end
-
+    sum = all.sum { |game| game.away_goals + game.home_goals}.to_f
+    (sum / all.length.to_f).round(2)
+  end
+
+  #MODULE!
+    def self.hash_of_hashes(collection, key1, key2, key3, value2, value3, arg2 = nil )
+      # {key1 => { key2 => value2, key3 => value3}} across collection. arg2 is ignored if not needed
+      # arg2 is optional if the passed method requires arguments
+      hash_of_hashes = Hash.new { |hash, key| hash[key] = {key2 => 0, key3 => 0}}
```

Comment:
this helper method creates a hash of hashes. It is a basic building block for nearly all my methods.
it creates the following: `{key1 => { key2 => value2, key3 => value3}}`
ex: goals per game across all seasons: 
```
{20132014=> { goals => 10, games_played => 5}
20142015=> { goals => 8, games_played => 3}}
```